[WIP] Update custom-builds.md (#1837)
cap-js/docs: guides/deployment/custom-builds.md

- The _src_ folder configured for the individual build task.
- If [feature toggles](../extensibility/feature-toggles#enable-feature-toggles) are enabled: subfolders of the folder _fts_.
- CDS Model folders and files defined by the [required services](../../node.js/cds-env#services) of your project. This also includes models used by required built-in CDS services, like [persistent outbox](../../node.js/outbox#persistent-outbox) or [MTX related services](../multitenancy/mtxs#mtx-services-reference).

Feature toggle folders and required built-in service models will also be added if user defined models have already been configured as [_model_ option](#build-task-properties) in your build tasks.

[Learn more about the calculation of the concrete list of CDS models.](../../node.js/cds-compile#cds-resolve){.learn-more}

::: tip If custom build tasks are configured, those properties have precedence
For example, you want to configure the _src_ folder and add the default models. To achieve this, do not define the _model_ option in your build task:

::: code-group

> ```json [package.json]

Reviewer: To not have the comment highlighted in red:

```suggestion
```jsonc [package.json]
```

{
  "build": {
    "target": "gen",
    "tasks": [
      {
        "for": "nodejs",
        "src": "srv",
        "options": { /* no "model" entry here */ }
      }
    ]
  }
}
```

> :::

Reviewer 2: I guess it should look like this, right?
![image](https://github.com/user-attachments/assets/cdc16609-85f4-4e9b-abec-9d1ae79f2f94)

```suggestion
```

Reviewer 2: Currently it looks like that:
![image](https://github.com/user-attachments/assets/bd21ed22-eea4-436e-9a92-af453d7b2996)

Author: good catch! :)

 That way, the model paths will still be dynamically determined, but the _src_ folder is taken from the build task configuration. So you benefit from the automatic determination of models, for example, when adding a new external services, or when CAP is changing any built-in service configuration values.
:::

> To control which tasks `cds build` executes, you can add them as part of your [project configuration](../../node.js/cds-env#project-settings) in _package.json_ or _.cdsrc.json_, as outlined [below](#build-task-properties).

Reviewer 2: ```suggestion
To control which tasks `cds build` executes, you can add them as part of your [project configuration](../../node.js/cds-env#project-settings) in _package.json_ or _.cdsrc.json_, as outlined [in the following chapter](#build-task-properties).
```

### Build Target Folder {#build-target-folder}

> If you want to change the default target folder, use the <Config keyOnly>cds.build.target=myfolder</Config> property. It is resolved based on the root folder of your project.

Reviewer: No idea...

> If you want to change the default target folder, use the <Config keyOnly>cds.build.target=myfolder</Config> property. It is resolved based on the root folder of your project.

Reviewer 3: Spell checkers don't like compound words like `myfolder`, so you could use `cds.build.target=my-folder` or something like `cds.build.target=path/to/my/folder`.

> If you want to change the default target folder, use the <Config keyOnly>cds.build.target=myfolder</Config> property. It is resolved based on the root folder of your project.

Author: seems to work, thanks, @swaldmann! 😃

}
```

:::

#### npm Workspace Support <Beta /> {#build-ws}

Use CLI option `--ws-pack` to enable tarball based deployment of [npm workspace](https://docs.npmjs.com/cli/using-npm/workspaces) dependencies. Workspaces are typically used to manage multiple local packages within a singular top-level root package.  Such a setup is often referred to as a [monorepo](https://earthly.dev/blog/npm-workspaces-monorepo/).

As an effect, your workspace dependencies can be deployed to SAP BTP without them being published to an npm registry before.

Behind the scenes, `cds build --ws-pack` creates a tarball in folder _gen/srv_ for each workspace dependency of your project that has a `*` version identifier.  Dependencies in _gen/package.json_ will be adapted to point to the correct tarball file URL:

::: code-group

> ```json [package.json]

Reviewer: ```suggestion
```jsonc [package.json]
```
